bazel/repositories.bzl

```diff
@@ -1,532 +1,150 @@
+            fail("External dep build failed")
+
+def envoy_dependencies(path = "@envoy_deps//", local_protobuf_bzl = None):
+    # Used only for protobuf.bzl.
+    if local_protobuf_bzl:
+        native.new_local_repository(
+            name = "protobuf_bzl",
+            path = "/thirdparty/protobuf-3.2.0",
+            # We only want protobuf.bzl, so don't support building out of this repo.
+            build_file_content = "",
+        )
+    else:
+        native.new_git_repository(
+            name = "protobuf_bzl",
+            # Using a non-canonical repository/branch here. This is a workaround to the lack of
+            # merge on https://github.com/google/protobuf/pull/2508, which is needed for supporting
+            # arbitrary CC compiler locations from the environment. The branch is
+            # https://github.com/htuch/protobuf/tree/v3.2.0-default-shell-env, which is the 3.2.0
+            # release with the above mentioned PR cherry picked.
+            commit = "d490587268931da78c942a6372ef57bb53db80da",
+            remote = "https://github.com/htuch/protobuf.git",
+            # We only want protobuf.bzl, so don't support building out of this repo.
+            build_file_content = "",
+        )
+
+    # Set this to True to make the build debug cycles faster.
+    debug_build = False
+
+    envoy_repository = repository_rule(
+        implementation = _repository_impl,
+        local = debug_build,
+        environ = ["CC", "CXX", "LD_LIBRARY_PATH"],
+        attrs = {"debug": attr.bool(default=False)},
     )
 
-def googletest_repositories():
-    BUILD = """
-cc_library(
-    name = "googletest",
-    srcs = [
-        "googlemock/src/gmock-all.cc",
-        "googletest/src/gtest-all.cc",
-    ],
-    hdrs = glob([
-        "googlemock/include/**/*.h",
-        "googlemock/src/*.cc",
-        "googletest/include/**/*.h",
-        "googletest/src/*.cc",
-        "googletest/src/*.h",
-    ]),
-    includes = [
-        "googlemock",
-        "googlemock/include",
-        "googletest",
-        "googletest/include",
-    ],
-    visibility = ["//visibility:public"],
-)
-"""
-    native.new_git_repository(
-        name = "googletest_git",
-        build_file_content = BUILD,
-        # v1.8.0 release
-        commit = "ec44c6c1675c25b9827aacd08c02433cccde7780",
-        remote = "https://github.com/google/googletest.git",
+    envoy_repository(
+        name = "envoy_deps",
+        debug = debug_build,
     )
 
-def http_parser_repositories():
-    BUILD = """
-cc_library(
-    name = "http_parser",
-    srcs = [
-        "http_parser.c",
-    ],
-    hdrs = [
-        "http_parser.h",
-    ],
-    visibility = ["//visibility:public"],
-)
-"""
-
-    native.new_git_repository(
-        name = "http_parser_git",
-        remote = "https://github.com/nodejs/http-parser.git",
-        commit = "9b0d5b33ebdaacff1dadd06bad4e198b11ff880e",
-        build_file_content = BUILD,
+    native.bind(
```

Comment:
I've added a skip_includes list parameter that can be used to exclude some binds.